Automattic/wp-calypso
# Add tier and price info to the Jetpack Search block on the plans page

#### Changes proposed in this Pull Request

* Makes the Jetpack Search block on the `/plans/[site]` and `jetpack/connect/...` pages show information about the site's search tier and number of search records (and the corresponding price that the user will be charged if they go through and buy it).
* A large part of the code here is for querying a new `sites/[site]/products` endpoint on the server, so that Calypso can obtain the site-specific price and tier info for Jetpack Search.  This code is based pretty heavily on the existing code in Calypso for querying the `sites/[site]/plans` endpoint.

#### Testing instructions

1. D38944-code is required on the server side to test this.  (Portions of that will need to be deployed before this can be merged.)
2. Go to `/plans` and select an already-connected Jetpack site. You should see something like this for the Jetpack Search block (the arrow shows the key part -- the info in this section should reflect your actual search usage as determined by the server):
![jetpack-search](https://user-images.githubusercontent.com/235183/77112779-fec32400-69ff-11ea-9f01-d6ab3dea64f0.png)
3. You can also connect a new Jetpack site, and within the plan selection page there you should see similar information:
![jetpack-search-connect](https://user-images.githubusercontent.com/235183/77113118-9759a400-6a00-11ea-98d7-2b69790a49ee.png)

Note that the design and wording here aren't fully finalized, but I based the text in current version on some internal mockups at p6TEKc-3jA-p2.
